src/fat.rs

```diff
@@ -12,11 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use core::convert::TryFrom;
+
```

Comment:
nit: Put all use statements together (old code might not but for new code this would be good.)